build_product_images.py

```diff
@@ -149,8 +192,164 @@ def product_to_build(product_name, product_version, products):
+    """
+    Builds neccessary dependencies for images if not available on system
+    """
+
+    args_dummy = copy.deepcopy(args)
+
+    if not rust_builder:
+        print("Building rust builder")
+        subprocess.run("make")
+
+    if not java:
+        args_dummy.image_version = '0'
+        args_dummy.product_version = '11'
+        print('Building dependency Java-Base', args_dummy.product_version)
+
+        run_commands(args_dummy.dry, build_and_publish_image(args_dummy, product_to_build('java-base', args_dummy.product_version, products)))
+
+        args_dummy.image_version = '0'
+        args_dummy.product_version = '1.8.0'
+        print('Building dependency Java-Base', args_dummy.product_version)
+
+        run_commands(args_dummy.dry, build_and_publish_image(args_dummy, product_to_build('java-base', args_dummy.product_version, products)))
+
+    if not tools:
+        args_dummy.image_version = '0'
+        args_dummy.product_version = '0.2.0'
+        print("Building dependency Tools", args_dummy.product_version)
+
+        run_commands(args_dummy.dry, build_and_publish_image(args_dummy, product_to_build('tools', args_dummy.product_version, products)))
+
+
+def check_platform(architecture):
+    """
+    Checks if a desired platform is given, gives current platform if not
+    """
+    if architecture is None:
+        architecture = platform.machine()
+
+    return architecture
+
+
+def create_virtual_enviroment(args):
+
+    commands = []
+
+    commands.append(
+        [
+            "docker",
+            "buildx",
+            "create",
+            "--name",
+            "builder",
+            "--use"
+        ]
+    )
+    run_commands(args.dry, commands)
+
+
+def remove_virtual_enviroment(args):
+
+    commands = []
+    commands.append(
+        [
+            "docker",
+            "buildx",
+            "rm",
+            "builder"
+        ]
+    )
+    run_commands(args.dry, commands)
+
+
+def join_nodes(args):
+
+    nodes = args.node.split(',')
+
+    for node in nodes:
+        commands = []
+        if node != nodes[0]:
+            commands.append(
+                [
+                    "docker",
+                    "buildx",
+                    "create",
+                    "--name",
+                    nodes[0],
+                    "--append",
+                    node
+                ]
+            )
+            run_commands(args.dry, commands)
+
+    use_builder(args)
+
+
+def use_builder(args):
+
+    nodes = args.node.split(',')
+    commands = []
+
+    commands.append(
+        [
+            "docker",
+            "buildx",
+            "use",
+            nodes[0]
+        ]
+    )
+
+    run_commands(args.dry, commands)
+
+
 def main():
     args = parse_args()
+    print("Current Platform: ", platform.machine())
+
+    if args.check:
+        check_or_build_dependencies(args, check_platform(args.architecture), conf.products)
```

Comment:
I would remove this for multiple reasons:
- the versions of the requirements are hard coded in this script.
- it introduces an additional dependency `import docker` which needs to be installed everywhere.
- this script is not a dependency manager

Reply:
Done